op-program/preimage/hints.go

```diff
@@ -0,0 +1,65 @@
+package preimage
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// HintWriter writes hints to an io.Writer (e.g. a special file descriptor, or a debug log),
+// for a pre-image oracle service to prepare specific pre-images.
+type HintWriter struct {
+	w io.Writer
+}
+
+var _ Hinter = (*HintWriter)(nil)
+
+func NewHintWriter(w io.Writer) *HintWriter {
+	return &HintWriter{w: w}
+}
+
+func (hw *HintWriter) Hint(v Hint) {
+	hint := v.Hint()
+	hintBytes := make([]byte, 4, 4+len(hint)+1)
+	binary.BigEndian.PutUint32(hintBytes[:4], uint32(len(hint)))
```

Comment:
There's a lot of `4` as a magic number. Is there a constant for uint32 bytes length we could use somewhere? Otherwise probably worth defining one here just to reduce the cognitive load for understanding what's happening.

Could potentially even do:
```
const uint32Size = 4
const hintOverheadSize = uint32Size + 1 // one extra byte for the trailing 0 byte
func (hw *HintWriter) Hint(v Hint) {
	hint := v.Hint()
	hintBytes := make([]byte, uint32Size, len(hint)+hintOverheadSize)
	binary.BigEndian.PutUint32(hintBytes[:uint32Size], uint32(len(hint)))
	hintBytes = append(hintBytes, []byte(hint)...)
	hintBytes = append(hintBytes, 0) // to block writing on
	_, err := hw.w.Write(hintBytes)
	if err != nil {
		panic(fmt.Errorf("failed to write pre-image hint: %w", err))
	}
}
```

Reply:
Or if we're ok with a slightly less optimised version:
```
func (hw *HintWriter) Hint(v Hint) {
	hint := v.Hint()
	var hintBytes []byte
	hintBytes = binary.BigEndian.AppendUint32(hintBytes, uint32(len(hint)))
	hintBytes = append(hintBytes, []byte(hint)...)
	hintBytes = append(hintBytes, 0) // to block writing on
	_, err := hw.w.Write(hintBytes)
	if err != nil {
		panic(fmt.Errorf("failed to write pre-image hint: %w", err))
	}
}
```